integration/mesos/src/main/java/tachyon/mesos/TachyonFramework.java

```diff
@@ -164,14 +164,24 @@ public void resourceOffers(SchedulerDriver driver, List<Protos.Offer> offers) {
           targetCpu = workerCpu;
           targetMem = workerMem;
           mWorkers.add(offer.getHostname());
+          mTaskName = sConf.get(Constants.INTEGRATION_MESOS_TACHYON_WORKER_NAME);
         } else {
           // The resource offer cannot be used to start either master or a worker.
+          driver.declineOffer(offer.getId());
           continue;
         }
+        
+        Protos.TaskID taskId =
+                Protos.TaskID.newBuilder().setValue(mTaskName + mLaunchedTasks).build();
```

Comment:
if you record the master task id as suggested below, you will be able to use mLaunchedTasks instead